src/main/java/org/mule/extensions/vm/internal/operations/VMOperations.java

```diff
@@ -0,0 +1,200 @@
+   * to be supplied on a temporal reply-To queue that this operation automatically creates.
+   * <p>
+   * The temporal reply queue is automatically disposed after a response is received or the timeout expires.
+   * <p>
+   * The queue on which the content is published has to be one for which a {@code vm:listener} exists. Otherwise, a
+   * {@code VM:QUEUE_NOT_FOUND} error is thrown.
+   *
+   * @param content         the content to be published
+   * @param queueDescriptor the queue configuration
+   * @param connection      the acting connection
+   * @param location        this processor's location
+   * @return The response generated from the invoked listener's flow
+   */
+  @Throws(PublishConsumeErrorTypeProvider.class)
+  public Result<Serializable, VMMessageAttributes> publishConsume(@Content TypedValue<Serializable> content,
+                                                                  @ParameterGroup(name = "queue") QueueDescriptor queueDescriptor,
+                                                                  @Connection VMConnection connection,
+                                                                  ComponentLocation location) {
+
+    final Queue queue = getQueue(queueDescriptor, connection, location, "publishConsume");
+    final Queue replyToQueue = queueManager.createReplyToQueue(queue, connection);
+
+    try {
+      doPublish(new ReplyToCommand(content, replyToQueue.getName()), queueDescriptor, queue);
+      return doConsume(replyToQueue, queueDescriptor)
+          .map(value -> asConsumeResponse(value, queueDescriptor))
+          .orElseThrow(() -> new ModuleException(format(
+                                                        "Published messages to queue '%s' but got no response after timeout of %d %s",
+                                                        queueDescriptor.getQueueName(), queueDescriptor.getQueueTimeout(),
+                                                        queueDescriptor.getQueueTimeoutUnit()),
+                                                 QUEUE_TIMEOUT));
+
+    } catch (ModuleException e) {
+      throw e;
+    } catch (Exception e) {
+      throw new MuleRuntimeException(createStaticMessage(format(
+                                                                "Found error trying to perform publish-consume to VM queue '%s'",
+                                                                queueDescriptor.getQueueName())),
+                                     e);
+    } finally {
+      queueManager.disposeReplyToQueue(replyToQueue);
+    }
+  }
+
+  private Result<Serializable, VMMessageAttributes> asConsumeResponse(Serializable value, QueueDescriptor queueDescriptor) {
+
+    Result.Builder<Serializable, VMMessageAttributes> resultBuilder = Result.builder();
+    resultBuilder.attributes(new VMMessageAttributes(queueDescriptor.getQueueName()));
+
+    if (value instanceof TypedValue) {
```

Comment:
seen this two times already, maybe is better to add this logic in the ResultBuilder?